vsintegration/src/FSharp.ProjectSystem.Base/Project/ProjectFactory.cs

```diff
@@ -146,6 +203,67 @@ protected override object PreCreateForOuter(IntPtr outerProjectIUnknown)
             return node;
         }
 
+        internal Microsoft.Build.Evaluation.Project ReinitializeMsBuildProject(string filename)
```

Comment:
The contents of this function look identical to the block of code above.  Refactor to reduce duplication.